kolibri/plugins/learn/assets/src/state/mutations.js

```diff
@@ -0,0 +1,31 @@
+const Vue = require('vue');
+const coreStore = require('kolibri.coreVue.vuex.store');
+
+const learnMutations = {
+  SET_PAGE_NAME(state, name) {
+    state.pageName = name;
+  },
+  SET_PAGE_STATE(state, pageState) {
+    state.pageState = pageState;
+  },
+  SET_EXAM_LOG(state, examLog) {
+    state.examLog = examLog;
+  },
+  SET_EXAM_ATTEMPT_LOGS(state, examAttemptLogs) {
+    Object.keys(examAttemptLogs).forEach(contentId => {
+      if (!state.examAttemptLogs[contentId]) {
+        Vue.set(state.examAttemptLogs, contentId, {});
+      }
+      Vue.set(state.examAttemptLogs, contentId,
```

Comment:
A little wary of refactor this late in the development cycle - would be good to avoid this sort of thing in future when we are heading towards cutting a beta.